### PR TITLE
Migrate to the drone-plugin-lib

### PR DESCRIPTION
**Pull request by donny-dont**
Just modifies the sources to use drone-plugin-lib and follow the boilr-template
for the organization. No major refactor was attempted. All code was just slotted
where drone-plugin-lib dictated.

### CODE REVIEW

File: plugin/impl.go [hunk 1]
```diff
@@ -9,107 +15,123 @@ import (
...
+func (p *Plugin) Validate() error {
+	if len(p.settings.Token) == 0 {
+		return fmt.Errorf("you must provide your drone access token")
 	}
 
-	p.Server = getServerWithDefaults(p.Server, p.Host, p.Proto)
-	if len(p.Server) == 0 {
-		return fmt.Errorf("Error: you must provide your Drone server.")
+	p.settings.server = getServerWithDefaults(p.settings.Server, p.settings.Host, p.settings.Proto)
+	if len(p.settings.server) == 0 {
```
Comment (tboerger) [inline plugin/impl.go:47]: This check doesn't make sense, if no server is provided the p.settings.Server variable will be anyway empty.

File: plugin/impl.go [hunk 2]
```diff
@@ -9,107 +15,123 @@ import (
...
 					" or for branch deploy last_successful should be true," +
 					" format repository@build/branch")
 			}
 		}
 
 		waiting := false
 
-		timeout := time.After(p.Timeout)
+		timeout := time.After(p.settings.Timeout)
+		//lint:ignore SA1015 refactor later
```
Comment (tboerger) [inline plugin/impl.go:112]: Instead of adding and forgetting this it could be directly replaced by NewTicker.

File: plugin/impl.go [hunk 3]
```diff
@@ -9,107 +15,123 @@ import (
...
 		tick := time.Tick(1 * time.Second)
 
+		var err error
```
Comment (tboerger) [inline plugin/impl.go:115]: Did I miss something or is this not required?

Comment (donny-dont) [reply plugin/impl.go:115]: Originally the plugin had an `Execute` that had all the logic. When the validation was split into `Validate` the `Execute` bits did not have `err` declared. So yea its necessary.

File: plugin/impl.go (hunk 2 above)
Comment (donny-dont) [reply plugin/impl.go:107]: I was expecting this to heavily change in support of https://github.com/drone-plugins/drone-downstream/issues/68 so I just added the ignore. If you want it fixed I can do it but I'm not sure how much of this code will remain afterwards.

Comment (tboerger) [reply plugin/impl.go:107]: time.Tick just got to be replaced by time.NewTicker, so it really doesn't involve any big functional change.

**Pull request merged**